Normalize log format

dgreene-r7 opened the pull request:
This PR normalizes logging format so that it can be parsed by [Logstash](https://www.elastic.co/products/logstash) or other log consumers.

fmitchell-r7 commented:
Thanks for fixing this!

Looking at the changes, I'm not sure which way the normalization went. Can you provide a sample of the old logging format(s) vs. the new logging format?

dgreene-r7 commented:
Sure can!

Old:
```
2016-04-13T20:56:46.194Z - info: Updated source consul source=consul, type=consul, name=rabbitmq, cluster=rabbitmq
2016-04-13T20:56:46.195Z - info: Updated source consul source=consul, type=consul, name=foo, cluster=foo
2016-04-13T20:56:46.197Z - info: Updated source consul source=consul, type=consul, name=bar, cluster=bar
2016-04-13T20:56:46.199Z - info: Updated source consul source=consul, type=consul, name=baz, cluster=baz
2016-04-13T20:56:47.509Z - verbose: {"remote_addr":"127.0.0.1","remote_user":"-","date":"13/Apr/2016:20:56:47 +0000","method":"GET","url":"/v1/conqueso","http_version":"1.1","status":"200","result_length":"-","referrer":"-","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.112 Safari/537.36","response_time":"8.626"}

2016-04-13T20:56:47.764Z - info: Updated source s3-config source=s3-config, type=s3
```

New:

```
{
  "source": "consul",
  "type": "consul",
  "name": "foo",
  "cluster": "foo",
  "level": "info",
  "message": "Updated source consul",
  "timestamp": "2016-04-13T20:59:09.546Z"
}
{
  "source": "consul",
  "type": "consul",
  "name": "bar",
  "cluster": "bar",
  "level": "info",
  "message": "Updated source consul",
  "timestamp": "2016-04-13T20:59:09.573Z"
}
{
  "res": {
    "statusCode": 200
  },
  "req": {
    "url": "/v1/conqueso",
    "headers": {
      "host": "localhost:9100",
      "connection": "keep-alive",
      "cache-control": "max-age=0",
      "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      "upgrade-insecure-requests": "1",
      "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.112 Safari/537.36",
      "dnt": "1",
      "accept-encoding": "gzip, deflate, sdch",
      "accept-language": "en-US,en;q=0.8",
      "cookie": ""
    },
    "method": "GET",
    "httpVersion": "1.1",
    "originalUrl": "/v1/conqueso",
    "query": {}
  },
  "responseTime": 6,
  "source": "request",
  "type": "request",
  "level": "info",
  "message": "GET /v1/conqueso 200 6ms",
  "timestamp": "2016-04-13T20:59:10.754Z"
}
```

The common fields to extract would be: `source`, `type`, `name`, `cluster`, `level`, `message`, and `timestamp`.

fmitchell-r7 commented:
Do the records pretty print like your example shows, or do they each come out on one line like this?
~~~json
{"source": "consul","type": "consul","name": "foo","cluster": "foo","level": "info","message": "Updated source consul","timestamp": "2016-04-13T20:59:09.546Z"}
{"source": "consul","type": "consul","name": "bar","cluster": "bar","level": "info","message": "Updated source consul","timestamp": "2016-04-13T20:59:09.573Z"}
~~~

dgreene-r7 commented:
Good call, @fmitchell-r7. I just checked back through Winston's docs and it turns out that the `json` options gives multi-line output and needs to be paired with the `stringify` option to force single-line.

See: https://github.com/winstonjs/winston/blob/master/docs/transports.md#console-transport

fmitchell-r7 commented on bin/server.js:
```diff
@@ -33,12 +35,17 @@ global.Config.defaults(require('../config/defaults.json'));
+app.use(expressWinston.logger({
+  winstonInstance: global.Log,
+  expressFormat: true,
+  level: global.Config.get('log:level'),
```
If we want to keep backwards compatibility, where access logs can be logged at a different level, should we set this to `global.Config.get('log:access:level')`?

fmitchell-r7 commented:
:+1: We'll need to remember to bump the minor version for the next release and make a note of the new log format in the release notes. :shipit:
